bot/utilities/Queue.ts

```diff
@@ -48,4 +50,45 @@ export class Queue {
     this.properties = { id, ownerId, channelId, startTime, status };
     this.entries = [];
   }
+
+  get length(): number {
+    return this.entries.length;
+  }
+  
+  checkQueue(context: TurnContext): boolean {
+    if (this.entries.find((student) => student.userId == context.activity.from.id) != undefined) {
+      return true;
+    }
```

Comment:
I think it would be cleaner to pass the `id` directly as a parameter here rather than passing the entire context object.

Reply:
Gotcha, I can change that. My initial thinking was that by passing context it would allow for easier changes. But I can see it only increases coupling.